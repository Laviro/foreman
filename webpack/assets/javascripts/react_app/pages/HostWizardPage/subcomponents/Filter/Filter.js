import React from 'react';
import PropTypes from 'prop-types';
import { Grid, FormGroup, FormControl, noop } from 'patternfly-react';

const Filter = ({ onChange }) => (
  <Grid.Row>
    <Grid.Col sm={4}>
      <FormGroup controlId="filter">
        <FormControl
          type="text"
          placeholder="Filter.."
          onChange={onChange}
          autoComplete="off"
        />
      </FormGroup>
    </Grid.Col>
  </Grid.Row>
);

Filter.propTypes = {
  onChange: PropTypes.func,
};

Filter.defaultProps = {
  onChange: noop,
};

export default Filter;
