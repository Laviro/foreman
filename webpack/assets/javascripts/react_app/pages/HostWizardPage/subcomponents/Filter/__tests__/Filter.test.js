import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import Filter from '../Filter';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('Filter', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(Filter, fixtures));
});
