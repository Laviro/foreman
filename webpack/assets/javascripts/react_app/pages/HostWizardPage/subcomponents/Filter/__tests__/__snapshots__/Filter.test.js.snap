// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter rendering render without Props 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    sm={4}
  >
    <FormGroup
      bsClass="form-group"
      controlId="filter"
    >
      <FormControl
        autoComplete="off"
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        placeholder="Filter.."
        type="text"
      />
    </FormGroup>
  </Col>
</Row>
`;
